superfine.d.ts

```diff
@@ -6,10 +6,12 @@ export type Children = VNode | string | number | null
  * The virtual DOM representation of an Element.
  */
 export interface VNode<Props = {}> {
-  name: string
-  props?: Props
```

Comment:
@tscholl2 Does Props? means Props are optional? If that's so, then it shouldn't be there.

Reply:
`props?: Props` means that the `props` property may or may not exist on a `VNode`, and if it does it exist, its type will be `Props`.

The reason I got rid of the `?` is that when I looked at the code, I thought that if a `VNode` is created through the `h` function, the `props` property is always set on `VNode`, and moreover, it is not null/undefined. I got that idea from this line:

https://github.com/jorgebucaran/superfine/blob/master/src/index.js#L422

Let me know if I am misunderstanding the code, or if this does not match the intended use of `VNode`. For example, if you intend other people to create `VNode`s through their own process and you want to convey that they do not need to set `props` on a `VNode` for the patch function to work (I didn't check to see if `patch` assumes the existence of this property).

Reply:
@tscholl2 I misunderstood the diff (yes, now that's a thing). I thought you added a `?`, but you _removed it_, which is 100% correct. Sorry! 😅